addons/sale_stock/report/stock_forecasted.py

```diff
@@ -14,18 +14,20 @@ def _prepare_report_line(self, quantity, move_out=None, move_in=None, replenishm
             return line
 
         picking = move_out.picking_id
-        line['move_out'].update({
-            'picking_id' : {
-                'id' : picking.id,
-                'priority' : picking.priority,
-                'sale_id' : {
-                    'id' : picking.sale_id.id,
-                    'amount_untaxed' : picking.sale_id.amount_untaxed,
-                    'currency_id' : picking.sale_id.currency_id.read(fields=['id', 'name'])[0] if read else picking.sale_id.currency_id,
-                    'partner_id' : picking.sale_id.partner_id.read(fields=['id', 'name'])[0] if read else picking.sale_id.partner_id,
+        # if read is False, line['move_out'] is a stock.move record and will trigger a record update
+        if read:
+            line['move_out'].update({
+                'picking_id': {
+                    'id': picking.id,
+                    'priority': picking.priority,
+                    'sale_id': {
+                        'id': picking.sale_id.id,
+                        'amount_untaxed': picking.sale_id.amount_untaxed,
+                        'currency_id': picking.sale_id.currency_id.read(fields=['id', 'name'])[0] if read else picking.sale_id.currency_id,
+                        'partner_id': picking.sale_id.partner_id.read(fields=['id', 'name'])[0] if read else picking.sale_id.partner_id,
```

Comment:
The `update` is inside the `if read` condition, so those conditions will always be true